tools/generate_refresh_template_spec.rb

```diff
@@ -0,0 +1,91 @@
+describe(ManageIQ::Providers::Redhat::InfraManager::Refresh::Refresher) do
+  ###
+  # This is not a real spec, it is used to generate a template for a refresh spec
+  # when we have a refresh that is known to work correctly and we need to test a
+  # new one. So we first run this spec with the working refresh and it generates a new spec
+  # file based on the template provided in the "filename" var and writes in to "output_file_name"
+  ###
+  before(:each) do
+    _guid, _server, zone = EvmSpecHelper.create_guid_miq_server_zone
+    @ems = FactoryGirl.create(:ems_redhat, :zone => zone, :hostname => "localhost", :ipaddress => "localhost",
+                              :port => 8443)
+    @ems.update_authentication(:default => {:userid => "admin@internal", :password => "123456"})
+    @ems.default_endpoint.verify_ssl = OpenSSL::SSL::VERIFY_NONE
+    allow(@ems).to(receive(:supported_api_versions).and_return([3]))
+    stub_settings_merge(:ems => { :ems_redhat => { :use_ovirt_engine_sdk => false } })
+  end
+
+  def relations_tree_to_a(expected_tree)
+    h = {}
+    return h if expected_tree.empty?
+    expected_tree = sort_tree(expected_tree)
+    expected_tree.each do |obj, children|
+      key = [wrap_name_for_inspect(obj.class.name), obj.name] # [obj.class.name, obj.name]
+      key << {:hidden => true} if obj.respond_to?(:hidden) && obj.hidden
+      h[key] = relations_tree_to_a(children)
+    end
+    h
+  end
+
+  def wrap_name_for_inspect(name)
+    InspectWrapper.new(name)
+  end
+
+  class InspectWrapper
+    attr_reader :name
+    def initialize(name)
+      @name = name
+    end
+
+    def inspect
+      name
+    end
+  end
+
+  def sort_tree(tree)
+    return tree if tree.blank?
+
+    if tree.first.first.kind_of?(Array)
+      # sorting expected tree
+      tree.sort_by { |key, _children| [key[0].name,    key[1]] }
+    else
+      # sorting actual tree
+      tree.sort_by { |obj, _children| [obj.class.name, obj.name] }
+    end
+  end
+
+  def tree_a_to_s(tree_a)
+    tree_s = tree_a.to_s
+    tree_s.gsub!(/\]\s*?=>\s*?{\[/, "] => {\n[")
+  end
+
+  it("will generate a new cassete") do
+    VCR.use_cassette("#{described_class.name.underscore}_generated_from_v3", :allow_unused_http_interactions => true, :allow_playback_repeats => true, :record => :new_episodes) do
```

Comment:
We should not commit allow_unused_http_interactions.

Reply:
done, and removed from the template as well.